fv3core/pace/fv3core/initialization/geos_wrapper.py

```diff
@@ -1,21 +1,76 @@
 import enum
+import logging
 import os
 from datetime import timedelta
 from typing import Dict, List, Tuple
 
 import f90nml
 import numpy as np
+from gt4py.cartesian.config import build_settings as gt_build_settings
+from mpi4py import MPI
 
 import pace.util
 from pace import fv3core
 from pace.driver.performance.collector import PerformanceCollector
-from pace.dsl.dace import DaceConfig, orchestrate
+from pace.dsl.dace import orchestrate
+from pace.dsl.dace.build import set_distributed_caches
+from pace.dsl.dace.dace_config import DaceConfig, DaCeOrchestration
 from pace.dsl.gt4py_utils import is_gpu_backend
 from pace.dsl.typing import floating_point_precision
 from pace.util._optional_imports import cupy as cp
 from pace.util.logging import pace_log
 
 
+class StencilBackendCompilerOverride:
+    """Override the Pace global stencil JIT to allow for 9-rank build
+    on any setup.
+
+    This is a workaround that requires to now _exactly_ when build is happening.
+    Using this as a context manager, we leverage the DaCe build system to override
+    the name and build the 9 codepaths required- while every other rank wait.
+
+    This should be removed when we refactor the GT JIT to distribute building
+    much more efficiently
+    """
+
+    def __init__(self, comm: MPI.Intracomm, config: DaceConfig):
+        self.comm = comm
+        self.config = config
+
+        # Orchestration or mono-node is not concerned
+        self.no_op = self.config.is_dace_orchestrated() or self.comm.Get_size() == 1
+
+        # We abuse the DaCe build system
+        if not self.no_op:
+            config._orchestrate = DaCeOrchestration.Build
+            set_distributed_caches(config)
+            config._orchestrate = DaCeOrchestration.Python
```

Comment:
What is happening here?

Reply:
So... I abuse the inner working of the DaCe system so I can re-use the cube localized directory name (.gt_cache_TL, T, TR...)

Refactoring the GT build for cube-sphere will extend this to no longer be a DaCe only thing